test/performance/odp_sched_pktio_run.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+#
+# Copyright (c) 2018, Linaro Limited
+# All rights reserved.
+#
+# SPDX-License-Identifier:	BSD-3-Clause
+#
+
+# directory where test binaries have been built
+TEST_DIR="${TEST_DIR:-$PWD}"
+# directory where test sources are, including scripts
+TEST_SRC_DIR=$(dirname $0)
+
+PATH=$TEST_DIR:$TEST_DIR/../../example/generator:$PATH
+
+# exit codes expected by automake for skipped tests
+TEST_SKIPPED=77
+
+VALIDATION_TESTDIR=platform/$ODP_PLATFORM/test/validation
+PLATFORM_VALIDATION=${TEST_SRC_DIR}/../../$VALIDATION_TESTDIR
+
+FLOOD_MODE=0
+
+# Use installed pktio env or for make check take it from platform directory
+if [ -f "./pktio_env" ]; then
+	. ./pktio_env
+elif  [ "$ODP_PLATFORM" = "" ]; then
+	echo "$0: error: ODP_PLATFORM must be defined"
+	# not skipped as this should never happen via "make check"
+	exit 1
+elif [ -f ${PLATFORM_VALIDATION}/api/pktio/pktio_env ]; then
+	. ${PLATFORM_VALIDATION}/api/pktio/pktio_env
+else
+	echo "BUG: unable to find pktio_env!"
+	echo "pktio_env has to be in current directory or "
+	echo "in platform/\$ODP_PLATFORM/test."
+	echo "ODP_PLATFORM=\"$ODP_PLATFORM\""
+	exit 1
+fi
+
+run_sched_pktio()
+{
+	setup_pktio_env clean # install trap to call cleanup_pktio_env
+
+	if [ $? -ne 0 ]; then
+		echo "setup_pktio_env error $?"
+		exit $TEST_SKIPPED
+	fi
+
+	type odp_generator > /dev/null
+	if [ $? -ne 0 ]; then
+		echo "odp_generator not installed. Aborting."
+		cleanup_pktio_env
+		exit 1
+	fi
+
+	# 1 worker
+	odp_sched_pktio${EXEEXT} -i $IF1,$IF2 -c 1 -s &
+
+	TEST_PID=$!
+
+	sleep 1
+
+	# Run generator with one worker
+	(odp_generator${EXEEXT} --interval $FLOOD_MODE -I $IF0 \
+			--srcip 192.168.0.1 --dstip 192.168.0.2 \
+			-m u -w 1 2>&1 > /dev/null) \
+			2>&1 > /dev/null &
+
+	GEN_PID=$!
+
+	# Run test for 5 sec
+	sleep 5
+
+	kill ${GEN_PID}
+	wait ${GEN_PID}
+
+	# Kill with SIGINT to output statistics
+	kill -2 ${TEST_PID}
+	wait ${TEST_PID}
+
+	ret=$?
+
+	if [ $ret -eq 3 ]; then
```

Comment:
Change C code to expect 0. Maybe  better to have argument how many packets you will feed from generator.

Reply:
Program returns by default 0, but when asked for statistics checking (-s) it returns check status: 0, 1,  2 or 3 (== success on both rx and tx). Return value is limited to max value 255, so cannot return packet count as return value.